app/Filament/Resources/EventResource/RelationManagers/EventAwardsRelationManager.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Filament\Resources\EventResource\RelationManagers;
+
+use App\Filament\Actions\ProcessUploadedImageAction;
+use App\Filament\Enums\ImageUploadType;
+use App\Models\Event;
+use App\Models\EventAward;
+use App\Models\User;
+use Filament\Forms;
+use Filament\Forms\Form;
+use Filament\Resources\RelationManagers\RelationManager;
+use Filament\Tables;
+use Filament\Tables\Table;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Auth;
+
+class EventAwardsRelationManager extends RelationManager
+{
+    protected static string $relationship = 'awards';
+
+    public static function canViewForRecord(Model $ownerRecord, string $pageClass): bool
+    {
+        /** @var User $user */
+        $user = Auth::user();
+
+        return $user->can('manage', EventAward::class);
+    }
+
+    public function form(Form $form): Form
+    {
+        /** @var Event $event */
+        $event = $this->getOwnerRecord();
+        $nextTier = ($event->awards()->max('tier_index') ?? 0) + 1;
+
+        return $form
+            ->schema([
+                Forms\Components\TextInput::make('label')
+                    ->minLength(2)
+                    ->maxLength(40)
+                    ->required(),
+
+                Forms\Components\TextInput::make('achievements_required')
+                    ->numeric()
+                    ->required(),
+
+                Forms\Components\TextInput::make('tier_index')
+                    ->default($nextTier)
+                    ->numeric()
+                    ->readOnly()
+                    ->required(),
+
+                Forms\Components\Section::make('Media')
+                    ->icon('heroicon-s-photo')
+                    ->schema([
+                        // Store a temporary file on disk until the user submits.
+                        // When the user submits, put in storage.
+                        Forms\Components\FileUpload::make('image_asset_path')
+                            ->label('Badge')
+                            ->disk('livewire-tmp') // Use Livewire's self-cleaning temporary disk
+                            ->image()
+                            ->rules([
+                                'dimensions:width=96,height=96',
+                            ])
+                            ->acceptedFileTypes(['image/jpeg', 'image/png', 'image/gif'])
+                            ->maxSize(1024)
+                            ->maxFiles(1)
+                            ->previewable(true),
```

Comment:
Looks like it's possible to submit without having uploaded a file. When doing so, `image_asset_path` is `EMPTY`. Should this field be set to required?

Reply:
It cannot be required for edit or the user has to upload a badge to change the label. I did figure out a way to make it required for new records.